zio-http/src/main/scala/zhttp/service/Client.scala

```diff
@@ -24,16 +24,16 @@ final case class Client(rtm: HttpRuntime[Any], cf: JChannelFactory[Channel], el:
   def request(
     request: Client.ClientParams,
     sslOption: ClientSSLOptions = ClientSSLOptions.DefaultSSL,
-  ): Task[Client.ClientResponse] =
-    for {
-      promise <- Promise.make[Throwable, Client.ClientResponse]
+  ): TaskManaged[Client.ClientResponse] =
+    (for {
+      promise <- Promise.make[Throwable, TaskManaged[Client.ClientResponse]]
```

Comment:
I think we can keep the promise as it is and call `close` in place of `toManaged_` below.
This will reduce the unnecessary "wrapping" of ZManaged and remove that additional "flatten" call